src/verify.c

```diff
@@ -867,21 +911,23 @@ int verify_main(int argc, char **argv)
 		if ((file_not_fixed_count == 0) && (file_not_replaced_count == 0)) {
 			remove_orphaned_files(official_manifest);
 		}
-		if (cmdline_option_picky) {
-			char *start = mk_full_filename(path_prefix, "/usr");
-			fprintf(stderr, "--picky removing extra files under %s\n", start);
-			ret = walk_tree(official_manifest, start, true);
+		if (cmdline_option_picky || cmdline_option_extra_picky) {
+			char *start = mk_full_filename(path_prefix, cmdline_option_extra_picky ? "" : "/usr");
+			fprintf(stderr, "%s removing extra files under %s\n", cmdline_option_extra_picky ? "--extra-picky" : "--picky", start);
+			ret = walk_tree(official_manifest, start, true,
+					cmdline_option_picky_whitelist_set ? &cmdline_option_picky_whitelist : NULL);
```

Comment:
regex_t is a struct and there is no documented way of determining whether it was initialized, at least as far as I can tell. So that has to be tracked separately to determine whether the a regex was chosen.

I used the same naming scheme as for the other existing variables. Should I use something shorter even if that then becomes inconsistent with the rest of the code? I don't have any strong opinion either way.

Reply:
I disagree, cmdline_option_picky_whitelist is nothing like the other existing variables. It is the result of compiling a regular expression which might have been given on the command line, or might be the default that is hard coded into the program. All the others are reflect if an option had been given on the command line. You could use cmdline_option_picky_whitelist as the name of the variable that holds the optarg value in the 'w' case, i.e. the command line textual representation of the regular expression, and something much shorter, e.g. whitelist_re for a pointer to the compiled version of that.

I would set up storage for the regex_t, and then where you are setting cmdline_option_picky_whitelist_set to true/false I would set whitespace_re to &storage or NULL.

I think a lot of the problem I have is I find the whole handling of `-w` un-intuitive. If you don't give it then you preserve some things like ${XXX}/usr/lib/{modules,kernel}, if you do give it then it is the list of patterns of things to preserve, unless it is empty, in which case it says don't preserve anything.  This special case of the empty pattern just looks weird to me. I know from decades of experience that empty regular expressions match everything, not nothing. If I wanted to match nothing I would give it a pattern like ^$ or /does/not/match/anything. This contradicts the help message, which should say something like "if the pattern is empty then it disables the whitelist feature which defaults to %s, otherwise it replaces the is a regular expression of things to whitelist"

Reply:
The variable is now called "picky_whitelist" and it's a pointer that gets checked for NULL.